docs/howtos/building-a-rust-component.md

```diff
@@ -1,20 +1,179 @@
+# Guide to Building a Rust Component
 
-## Guide to Building a Rust Component
-
-(Most likely you are looking for https://github.com/mozilla/application-services/blob/main/docs/building.md)
+This is a guide to creating a new Rust Component. If you are looking for
+information how to build (ie, compile, etc) the existing components,
+you are looking for [our build documentation](https://github.com/mozilla/application-services/blob/main/docs/building.md)
 
 Welcome!
 
-It's great that you want to build a Rust Component, but we haven't written this
-guide yet.
+It's great that you want to build a Rust Component - this guide should help
+get you started. It documents some nomenclature, best-practices and other
+tips and tricks to get you started.
+
+This document is just for general guidance - every component will be different
+and we are still learning how to make these components. Please update this
+document with these learnings.
+
+To repeat with emphasis - **please consider this a living document**.
+
+# General design and structure of the component
+
+We think components should be structured as described here.
+
+## The "store" is the "entry-point"
+
+[Note that some of the older components use the term "store" differently; we
+should rename them! In places, it's called an "API". See `webext-storage` for
+a more recent component that uses the term "Store" as we think it should be
+used.]
+
+The "Store" is the entry-point for your application - it will provide the
+top-level functionality exposed by the client, which manages your databases and
+another singletons. The responsibilities of this will include things like
+creating the DB if it doesn't exist, doing schema upgrades etc.
```

Comment:
This may be something you're planning to expound on, but I'd like more info on schema upgrades? Are there best practices? Should certain changes be avoided?

Reply:
I added a new section just for this 👍